### PR TITLE
Add Bomex box experiment

### PR DESCRIPTION
**Pull request by trontrytel**
This PR adds a bomex box experiment setup (without the sub-grid scale model). 

I think the initial condition and forcings are applied correctly. Not sure about the surface parameters...

### CODE REVIEW

File: post_processing/post_processing_funcs.jl [hunk 1]
```diff
@@ -130,6 +130,13 @@ function postprocessing_box(sol, output_dir)
         clim = (-0.1, 0.1),
     )
     Plots.png(p, joinpath(output_dir, "w.png"))
+    qt = Y.c.ρq_tot ./ Y.c.ρ
```
Comment (szy21) [inline post_processing/post_processing_funcs.jl:133]: This will cause a problem for the dry case. You can add a check for whether `ρq_tot` is in `Y.c`.

**Review by szy21 — APPROVED** [review]

File: post_processing/post_processing_funcs.jl (hunk 1 above)
Comment (trontrytel) [reply post_processing/post_processing_funcs.jl:133]: Right! thanks

**Comment by trontrytel** [comment]
Seems like the initial condition for `rhoq_tot` is not used - we have `q_tot = 0` at the end. I'll look into this.

**Comment by szy21** [comment]
Oh, I think it may be because [here](https://github.com/CliMA/ClimaAtmos.jl/blob/e8e5570cf4ddad394f55681024400ef42470a06b/src/examples_types.jl#L229) if the config is box then the dry initial condition will be used. However, even if we start dry the large-scale tendency in q_tot should result in non-zero q_tot in the end?

**Comment by trontrytel** [comment]
Right. Something else is suspicious. I'll see if I can disentangle the initial condition from the case configuration. And then take a look at the forcings again

File: src/examples_types.jl [hunk 2]
```diff
@@ -220,22 +220,19 @@ function get_state_restart(comms_ctx)
 end
 
 function get_initial_condition(parsed_args)
-    if is_baro_wave(parsed_args)
-        if parsed_args["moist"] == "dry"
-            return ICs.DryBaroclinicWave(parsed_args["perturb_initstate"])
+    if isnothing(parsed_args["turbconv_case"])
+        if parsed_args["initial_condition"] in ["DryBaroWave", "MoistBaroWave", "DecayingProfile"]
```
Comment (szy21) [inline src/examples_types.jl:224]:
```suggestion
        if parsed_args["initial_condition"] in ["DryBaroclinicWave", "MoistBaroclinicWave", "DecayingProfile"]
```

Comment (trontrytel) [reply src/examples_types.jl]: Must have forgotten to push the fix to this! Thank you.

**Pull request merged**